### PR TITLE
Feature/28 reconnect to room

### LINKED ISSUE #28: Add ability to reconnect to a room
Right now if a user needs to refresh for whatever reason they are able to rejoin but will lose any associations made to their userId. For example, if a moderator refreshes they will become a voter when they reconnect and a new moderator will be chosen. Or if a voter voted, but needs to refresh, their selection is lost since it's a "different" user.

### PR DESCRIPTION
**Pull request by misterbastean**
closes #28 

Added `userId` as a JWT stored as a cookie. 

To test, just hit the landing page and check your cookies and make sure you have `JWT_TOKEN` cookie. Then create and join a room with 2+ users. Refresh the page and you should still have the same cookie, etc.

If you refresh the page with the moderator, one of the other users will be promoted to moderator and the old moderator will be listed as a voter.

### CODE REVIEW

File: server/routes/fe.routes.ts
Comment (lvl-mattwells) [inline server/routes/fe.routes.ts:36]: You can simplify this conditional here and in the other route now 🙂

File: server/routes/fe.routes.ts
Comment (lvl-mattwells) [inline server/routes/fe.routes.ts:19]: Can we add `secure` so that it can't be modified?

File: server/controllers/socket.controller.ts
Comment (lvl-mattwells) [inline server/controllers/socket.controller.ts:242]: Are we handling the transfer of selection/confidence when a player rejoins anywhere? If I make a selection and refresh, do I lose my selection or am I able to keep it? I think that the main value of this ticket is that you can refresh and not lose any state.

**Review by lvl-mattwells — CHANGES_REQUESTED** [review]

File: server/controllers/socket.controller.ts
Comment (misterbastean) [reply server/controllers/socket.controller.ts:242]: Not sure, honestly, as the selection cards aren't in yet. Maybe I should hold off on this until we merge that one.

Comment (lvl-mattwells) [reply server/controllers/socket.controller.ts:242]: That PR just so happens to be ready for review #35 👀

File: server/controllers/socket.controller.ts
Comment (lvl-mattwells) [inline server/controllers/socket.controller.ts:18]: Would a `Map` make more sense here? Might be easier to manipulate with userId as the key.

File: server/controllers/socket.controller.ts
Comment (lvl-mattwells) [inline server/controllers/socket.controller.ts:67]: Do we need to send a room update if nothing changed?

File: server/controllers/socket.controller.ts
Comment (misterbastean) [reply server/controllers/socket.controller.ts:18]: Good call, implementing.

File: server/controllers/socket.controller.ts
Comment (misterbastean) [reply server/controllers/socket.controller.ts:67]: Yes, or else the user who reconnects doesn't get the info. There may be a better way to do this, but I haven't found it.

**Comment by lvl-mattwells** [comment]
Noticed some weirdness. Not exactly sure what's going on.
[ConnectionWeirdness.webm](https://user-images.githubusercontent.com/92868767/184449305-5de3587c-24fa-4b29-834b-dac85a30459f.webm)

**Comment by lvl-mattwells** [comment]
Closing in favor of #58

**Pull request closed without merging**
